Ginger/GingerCoreCommon/GeneralLib/ObservableList.cs

```diff
@@ -257,68 +259,60 @@ public IEnumerable<T> ItemsAsEnumerable()
         {
             return Items.AsEnumerable<T>();
         }
-
-        
-        string mStringData = null;
+                
         string mFilterStringData = null;
-        MemoryStream mMemoryStream = null;
-        int mDataLen;
-
+        
+       
 
         protected new IList<T> Items
         {
             get
             {
-                if (mLazyLoad)
+                if (LazyLoad)
                 {
                     LoadLazyInfo();
                 }
                 return base.Items;
             }
         }
 
-        bool mLazyLoad = false;
-        bool IObservableList.LazyLoad { get { return mLazyLoad; } set { mLazyLoad = value; } }
-        public bool LazyLoad { get { return mLazyLoad; } }
+        public LazyLoadListDetails LazyLoadDetails { get; set; }
+
+        bool IObservableList.LazyLoad 
+        { 
+            get { return LazyLoad; } 
+        }
+        public bool LazyLoad 
+        { 
+            get 
+            { 
+                if (LazyLoadDetails == null)
+                {
+                    return false;
+                }
+                else
+                {
+                    return (!LazyLoadDetails.DataWasLoaded);
+                }
+            } 
+        }
 
         bool mAvoidLazyLoad = false;
         public bool AvoidLazyLoad { get { return mAvoidLazyLoad; } set { mAvoidLazyLoad = value; } }
 
-        public string StringData { get { return mStringData; } set { mStringData = value; } }
-
-        public MemoryStream StringDataMS { get { return mMemoryStream; } set { mMemoryStream = value; } }
-        public int DataLen { get { return mDataLen; } set { mDataLen = value; } }
-
         public new IEnumerator<T> GetEnumerator()
         {
-            if (mLazyLoad)
+            if (LazyLoad)
             {
                 LoadLazyInfo();
             }
             return base.GetEnumerator();
         }
 
-        public void DoLazyLoadItem(string s)
-        {
-            //option 1 simple string 
-            StringData = s;
-
-            //Option 2 compressed string
-            // observableList.StringData = StringCompressor.CompressString(s);
-
-            //option 3 MS
-            //StringDataMS = StringCompressor.CompressStringToBytes(s);
-            // DataLen = s.Length;
-
-            mLazyLoad = true;
-
-        }
-
         bool loadingata = false;
-
         public void LoadLazyInfo()
         {
-            if (!mLazyLoad) return;
+            if (!LazyLoad) return;
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5848147)